tests/run-make/redundant-libs/rmake.rs

```diff
@@ -0,0 +1,28 @@
+// rustc will remove one of the two redundant references to foo below.  Depending
+// on which one gets removed, we'll get a linker error on SOME platforms (like
+// Linux).  On these platforms, when a library is referenced, the linker will
+// only pull in the symbols needed _at that point in time_.  If a later library
+// depends on additional symbols from the library, they will not have been pulled
+// in, and you'll get undefined symbols errors.
+//
+// So in this example, we need to ensure that rustc keeps the _later_ reference
+// to foo, and not the former one.
+
+//@ ignore-cross-compile
+// Reason: the compiled binary is executed
+
+//FIXME(Oneirical): msvc
```

Comment:
Suggestion: try checking `.dll.lib` for import library on msvc.

Reply:
```diff
-let lib_path = dynamic_lib_name(lib_name);
+let lib_path = if is_msvc() { format!("{lib_name}.dll.lib") } else { dynamic_lib_name(lib_name) };
```

Reply:
Okay, that was misguided, I'm trying to do the check in the test itself now.